app/src/main/java/pl/pwr/zpi/bcycle/mobile/DashboardActivity.kt

```diff
@@ -0,0 +1,133 @@
+package pl.pwr.zpi.bcycle.mobile
+
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import coil.api.load
+import com.google.firebase.auth.FirebaseAuth
+import kotlinx.android.synthetic.main.activity_dashboard.*
+import kotlinx.android.synthetic.main.row_future_trip.view.*
+import kotlinx.android.synthetic.main.row_history_trip.view.*
+import org.threeten.bp.format.DateTimeFormatter
+import org.threeten.bp.format.FormatStyle
+import pl.pwr.zpi.bcycle.mobile.api.ApiClient
+import pl.pwr.zpi.bcycle.mobile.models.GroupTrip
+import pl.pwr.zpi.bcycle.mobile.models.Trip
+import pl.pwr.zpi.bcycle.mobile.utils.background
+import pl.pwr.zpi.bcycle.mobile.utils.showToastError
+import kotlin.math.round
+
+class DashboardActivity : BCycleNavigationDrawerActivity() {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_dashboard)
+        setListeners()
+        arrangeRecyclerViews()
+    }
+
+    private fun arrangeRecyclerViews() {
+        val adapter = HistoryTripAdapter(mutableListOf<Trip>())
```

Comment:
Two recycler views are unintuitive, try merging the data in one RV. The adapter can use [`getItemViewType`](https://developer.android.com/reference/androidx/recyclerview/widget/RecyclerView.Adapter#getItemViewType(int)) and the `viewType` argument of `onCreateViewHolder` to support two (or more) styles of viewholders.

Also, groupTripApi returns all group trips, not just future trips. Those should be probably sorted by date when merging private and group trips.